soci/artifacts.go

```diff
@@ -199,6 +199,49 @@ func (db *ArtifactsDb) GetArtifactEntry(digest string) (*ArtifactEntry, error) {
 	return &entry, nil
 }
 
+// RemoveArtifactEntryByIndexDigest removes an index's artifact entry using its digest
+func (db *ArtifactsDb) RemoveArtifactEntryByIndexDigest(digest string) error {
+	return db.db.Update(func(tx *bolt.Tx) error {
+		bucket, err := getArtifactsBucket(tx)
+		if err != nil {
+			return err
+		}
+
+		return bucket.DeleteBucket([]byte(digest))
```

Comment:
This should check `indexBucket` before deleting, right?

Reply:
You're right, fixed.